CleverGirl/CleverGirl/Objects/CandidateWeapons.cs

```diff
@@ -113,8 +118,25 @@ public CandidateWeapons(AbstractActor attacker, ICombatant target)
                     Mod.Log.Info?.Write($"  -- attackAngle: {attackAngle}  positionDeltaAngle: {positionDeltaAngle}  deltaAngle: {deltaAngle}  absAngle: {absAngle}");
                     Mod.Log.Info?.Write($"  -- target pos: {target.CurrentPosition}  rot: {target.CurrentRotation}");
                 }
+            }
+        }
 
+        private static bool HasAvailableAmmoMode(Weapon weapon)
+        {
+            AmmoModePair currentAmmoMode = weapon.getCurrentAmmoMode();
+            bool canFire = false;
+            foreach (AmmoModePair ammoModePair in weapon.getAvaibleFiringMethods())
+            {
+                weapon.ApplyAmmoMode(ammoModePair);
+                if (weapon.CanFire)
```

Comment:
IIRC I think weapon.CanFire already checks the ammo count, but if not might be worth checking here?

Reply:
It does check the ammo count through a CAC transpile on HasAmmo inside the call to check the set ammomodes count, that is why I switch the ammomodes around.

Reply:
Great